tsup.config.ts

```diff
@@ -0,0 +1,48 @@
+import { defineConfig } from 'tsup'
+
+export default defineConfig([
+  {
+    name: 'index',
```

Comment:
I think we have a couple files in msw that are referenced from `interceptors/util` like https://github.com/mswjs/msw/blob/ab8b1accf5db10b30647759dde8c1585b8dc4c14/src/utils/matching/matchRequestUrl.ts#LL2C3-L2C3

do we need to expose those as well, or can we export them from the root index?

Reply:
They are already exported from the root index - I don't know if the reason for importing like this in msw was for tree-shaking purposes? If so we'll need to separate the utils into another bundle which is then re-exported from the index somehow

Reply:
cool - i'm not sure either, but if we can use them from another path that might be helpful - but it might be worth adding the entrypoint to be consistent for now, and change imports later if we need to!

Reply:
Looking at how msw imports these utils more closely, it seems to import from specific files like `interceptors/utils/getCleanUrl.js` and `interceptors/utils/bufferUtils.js`. There is no `utils/index.ts` file which we can use to bundle all the utils together.

Based on this, we would need to add an entrypoint for all the individual files under `utils/` which seems a bit unusual to me. Are we sure we want that? I'd like to get @kettanaito's opinion here as well

Reply:
Please, let's make those a part of the public API, exporting them from the root of the package. That was a historic oversight on my part and now seems like a good opportunity to finally fix it. Let's not reference things that way. 

So, what I'm proposing is this:

```
import { getCleanUrl, bufferUtils } from '@mswjs/interceptors'
```

Reply:
`getCleanUrl`, `encodeBuffer` and `decodeBuffer` are already part of the public API exported from the root of the package so no change is needed unless there are more utils that should be part of the public API